packages/earth-admin/src/pages-client/layers/details.tsx

```diff
@@ -14,133 +14,476 @@
+  const [layerType, setLayerType] = useState(null);
+  const [layerProvider, setLayerProvider] = useState(null);
+  const [toggle, setToggle] = useState(false);
+  const [copySuccess, setCopySuccess] = useState('');
+  const textAreaRef = useRef(null);
+
+  const {
+    id,
+    name,
+    description,
+    primary,
+    published,
+    version,
+    createdAt,
+    updatedAt,
+    slug,
+    type,
+    provider,
+    category,
+    config,
+    references,
+  } = layer;
 
-  const {selectedGroup, getPermissions} = useAuth0();
 
-  const permissions = getPermissions(AuthzGuards.accessLayersGuard);
+  useEffect(() => {
+    setLayer(data);
 
+  }, [data]);
 
-  const handleSearchValueChange = (newValue: string) => {
-    setPageCursor(INIT_CURSOR_LOCATION);
-    setNextCursor(null);
-    setSearchValue(newValue);
-  };
 
-  const handleCursorChange = () => {
-    if (nextCursor) {
-      setPageCursor(nextCursor);
-    }
-  };
+  useEffect(() => {
+    layer.config && setLayerConfig(layer.config);
+    layer.category && setLayerCategory(getSelectValues(LAYER_CATEGORY_OPTIONS, layer.category));
+    layer.type && setLayerType(LAYER_TYPE_OPTIONS.find((t) => t.value === layer.type));
+    layer.provider && setLayerProvider(LAYER_PROVIDER_OPTIONS.find((p) => p.value === layer.provider));
+  }, [layer]);
+
+
+  const {getValues, register, formState, errors, control} = useForm({
+    mode: 'onChange',
+  });
+
+  const {touched, dirty, isValid} = formState;
+  const renderErrorFor = setupErrors(errors, touched);
 
   useEffect(() => {
-    async function setupLayers() {
-      setIsLoading(true);
+    setFormValid(isValid);
+  }, [isValid]);
+
 
-      const query = {
-        search: searchValue,
-        sort: 'name',
-        page: {size: pageSize, cursor: pageCursor},
-        group: selectedGroup,
-      };
-      const encodedQuery = encodeQueryToURL('layers', query);
-      const res: any = await getAllLayers(encodedQuery);
+  async function onSubmit(e?, setIsEditing?, setIsLoading?, setServerErrors?) {
+    e.preventDefault();
 
+    const formData = getValues();
 
-      setTotalResults(res.total);
+    const {type, category, provider, references} = formData;
 
-      setLayer(!nextCursor ? res.data : [...layers, ...res.data]);
-      setNextCursor(res.pagination.nextCursor ? res.pagination.nextCursor : null);
-      setIsNoMore(!res.pagination.nextCursor);
+    const parsed = {
+      ...formData,
+      ...(category && {category: flattenArrayForSelect(category, 'value')}),
+      ...(type && {type: flattenObjectForSelect(type, 'value')}),
+      ...(provider && {provider: flattenObjectForSelect(provider, 'value')}),
+      ...(references && {references: flattenArrayForSelect(references, 'id')}),
+    };
 
-      setIsLoading(false);
+    try {
+      setIsLoading && setIsLoading(true);
+      await handleLayerForm(false, parsed, id, selectedGroup);
+      const res = await getLayer(encodedQuery);
```

Comment:
Is the extra get request needed? `handleLayerForm` seems to return the exact same data

Reply:
put request doesn't bring back included fields